guacamole-common-js/src/main/webapp/modules/Tunnel.js

```diff
@@ -1015,7 +1015,7 @@ Guacamole.WebSocketTunnel = function(tunnelURL) {
                     var opcode = elements.shift();
 
                     // Update state and UUID when first instruction received
-                    if (tunnel.state === Guacamole.Tunnel.State.CONNECTING) {
+                    if (tunnel.state === Guacamole.Tunnel.State.CONNECTING || tunnel.uuid === null) {
```

Comment:
If the issue is that the value of `tunnel.uuid` is being inadvertently overly coupled to tunnel state, and that the tunnel UUID should be accepted when sent via `INTERNAL_DATA_OPCODE` so long as it hasn't yet been set, perhaps this should just be `if (!tunnel.uuid)`?

Reply:
agreed. Do you prefer `!tunnel.uuid` over `tunnel.uuid === null`?

Reply:
Eh ... I have nothing against keeping `=== null` in this case. `tunnel.uuid` is documented specifically as being `null` if not known, and the WebSocket implementation can certainly be aware of its own use of `null`.